config/environments/production.php

```diff
@@ -11,6 +11,17 @@
 define('S3_UPLOADS_SECRET', env('S3_UPLOADS_SECRET'));
 define('S3_UPLOADS_REGION', env('S3_UPLOADS_REGION'));
 define('S3_UPLOADS_ENDPOINT', env('S3_UPLOADS_ENDPOINT'));
+define('S3_UPLOADS_BUCKET_URL', env('S3_UPLOADS_BUCKET_URL'));
+
+if (env('S3_UPLOADS_ENDPOINT') && env('S3_UPLOADS_BUCKET_URL') == '') {
```

Comment:
What if both `S3_UPLOADS_ENDPOINT` and `S3_UPLOADS_BUCKET_URL` are defined?

I actually find it really odd to have that many different environment variables to name the same thing. Isn't it possible to simplify?

Reply:
If both are defined, it's good. The customer has the choice to define `S3_UPLOADS_BUCKET_URL` or not. If it is not defined (by default), we build ourselves the complete bucket URL with the required information. 

I don't think it can be simplified any further here. It's the way the plugin works. It's better to build on that than to change everything knowing that there is documentation on this plugin here: https://github.com/humanmade/S3-Uploads